### DISCUSSION TITLE
Image showing in MD preview, not site. Paths/spelling are correct; strange behavior with `static`??

### DISCUSSION THREAD

**Discussion by cconn30** [community/community#75247] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Select Topic Area

Question

### Body

Over the last few days, I have been unable to get images to show up on my GitHub pages site, however they do show up on the remote repository in the markdown preview. I had all of my images in `static`, however the only one which did indeed show on the website was an image which is referenced in `_index.md`. I moved the remaining images (and PDFs, excluding the functioning image which remains in `static`) to `assets` as I saw recommended on another community post, but to no avail.

I have checked spelling/capitalization, about every filepath that I can think of, tried both HTML and markdown, and every other solution I've seen on the internet. In particular, I have tried using `../../assets/dm1.jpg` (for example) as suggested in a previous community discussion, and this changed nothing. I have also tried doing the same with a folder `/docs/assets`, as well as in `static`. I have no idea why a relatively referenced image in `static` works when called in `content/_index.md`, but not if called in `content/posts/photogallery.md` (with both relative and absolute file paths). Is this a bug, or is there something wrong with perhaps my `hugo.toml` configuration file (or similar)?

I can't figure this out, help would be massively appreciated! Thank you!

Note: the images which do not work are in `/content/post/photogallery.md`, and the functioning image is in `static`. 

repo: https://github.com/cconn30/ccxyz

**Comment by sky9262** [comment, marked as answer]
Hi @cconn30 ,

## There are two ways to access the image

### 1. From `static` folder
You can access the static images by following the code
```html
<img src="/dm1.jpg" alt="dartmouth 1" width="225">
```


### 2. From `assets` folder
In order to access files from assets, you can create a costume `shortcodes`.

- create `layouts/shortcodes/img.html` in the root folder
- write the following code in the `img.html`:
```html
{{ $image := resources.Get (.Get 0) }}
<img src="{{ $image.RelPermalink }}" alt="{{ (.Get 1) }}">
```
OR (if you want to resize)
```html
{{ $image := resources.Get (.Get 0) }}
<img src="{{ $image.RelPermalink }}" alt="{{ (.Get 1) }}" style="width: {{ (.Get 2) }}px; height: {{ (.Get 3) }}px;">
```
- Use the following code to render the image:
```markdown
{{< img "{image_name}" "{alt_text}" >}}
```
```markdown
#sample
{{< img "dm2" "dartmouth 2" >}}
```
OR (if you want to resize)
```markdown
{{< img "{image_name}" "{alt_text}" "{width_px}" "{heigth_px}" >}}
```
```markdown
#sample
{{< img "dm2" "dartmouth 2" "300" "500">}}
```
> Note: Don't forget to replace the `image_name` and `alt_text`
> Note: If your image is in `assets/image` the image_name should be `image/image_name.jpg`

Let me know if you need the full code!!!!!
